validator/client/attest.go

```diff
@@ -79,6 +73,32 @@ func (v *validator) SubmitAttestation(ctx context.Context, slot uint64, pubKey [
 		}
 		return
 	}
+	indexedAtt.Signature = sig
+	slashable, err := v.localSlashingProtector.IsSlashableAttestation(ctx, indexedAtt, pubKey, signingRoot)
+	if err != nil {
+		log.WithFields(
+			attestationLogFields(pubKey, indexedAtt),
+		).WithError(err).Error("Could not check attestation safety with slashing protection, not submitting")
+		return
+	}
+	if remoteProtector, ok := v.remoteSlashingProtector.(*remote.Service); ok && remoteProtector != nil {
```

Comment:
A future improvement here would be to do these things in parallel. I am assuming that the validator client is basically idle while waiting for a response from the remote slasher. During the time that the remote request is being fulfilled, the validator could also check the local db.